flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/JobSubmitRequestBody.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.rest.messages.job;
+
+import org.apache.flink.runtime.jobgraph.JobGraph;
+import org.apache.flink.runtime.rest.messages.RequestBody;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+import java.util.Arrays;
+
+/**
+ * Request for submitting a job.
+ *
+ * <p>We currently require the job-jars to be uploaded through the blob-server.
+ */
+public final class JobSubmitRequestBody implements RequestBody {
+
+	private static final String FIELD_NAME_SERIALIZED_JOB_GRAPH = "serializedJobGraph";
+
+	/**
+	 * The serialized job graph.
+	 */
+	@JsonProperty(FIELD_NAME_SERIALIZED_JOB_GRAPH)
+	public final byte[] serializedJobGraph;
```

Comment:
Is there a size limitation for what we can send via the REST client/server?

Reply:
the current max is 10mb.

Reply:
Alright, will a proper exception be thrown when his limit is exceeded? Otherwise we could check right away when creating this request.

Reply:
I know the the HttpObjectAggregator will throw an exception, and i _think_ it will just be logged server-side. We can't really change that except my modifying the aggregator (but i have some WIP to replace it anyway).

I'll add the client-side size check.